timeutil/schedule.go

```diff
@@ -28,7 +29,9 @@ import (
 	"time"
 )
 
-var validTime = regexp.MustCompile(`^([0-9]|0[0-9]|1[0-9]|2[0-3]):([0-5][0-9])$`)
+// Match 0:00-24:59, so that we cover 0:00-24:00, where 24:00 means the later
+// end of the day. 24:<mm> where mm > 0 must be handled separately.
+var validTime = regexp.MustCompile(`^([0-9]|0[0-9]|1[0-9]|2[0-4]):([0-5][0-9])$`)
```

Comment:
I think we should cut off the case of hour == 24 and minute > 0 here with the regexp, rather than below with `if t.Hour == 24` special case, i.e. `|24:00` in the regexp as the only valid alternative to the entire regexp (+ replace `2[0-4]` with `2[0-3]`.

Reply:
Agreed, using`24:00` explicitly looks like a better approach.